### PR TITLE
Track time (and inflight request count) spent waiting on storage sempahore limiter.

### PR DESCRIPTION
**Pull request by goral09**
Also changes the bucket resolution for trie-access methods. Previously started at 10ms and tracked until 10s, now starts at 1ms and trakcs until 1s.

### CODE REVIEW

File: node/src/components/storage.rs [hunk 1]
```diff
@@ -316,8 +328,12 @@ where
             // number of synchronous background tasks, especially since tokio does not put
             // meaningful bounds on these by default. For this reason, we limit the maximum number
             // of blocking tasks using a semaphore.
+            let start = Instant::now();
+            sync_task_limiter_inflight_counter.inc();
             let outcome = match sync_task_limiter.acquire().await {
                 Ok(_permit) =>{
+                    sync_task_limiter_inflight_counter.inc_by(-1.0);
```
Comment (Fraser999) [inline node/src/components/storage.rs]: Shouldn't we decrement in the error branch too?

File: node/src/components/storage/metrics.rs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            "storage_sync_task_limiter_in_flight_count",
+            "number of currently awaiting tasks on the semaphore",
+        )?;
+        registry.register(Box::new(sync_task_limiter_in_flight_counter.clone()))?;
+
+        Ok(Metrics {
+            sync_task_limiter_in_flight_counter,
+            sync_task_limiter_waiting_millis: utils::register_histogram_metric(
+                registry,
+                "storage_sync_task_limiter",
```
Comment (Fraser999) [inline node/src/components/storage/metrics.rs]: Maybe more explicit?
```suggestion
                "storage_sync_task_limiter_waiting_millis",
```

File: node/src/components/storage.rs [hunk 3]
```diff
@@ -360,7 +377,9 @@ impl Storage {
         finality_threshold_fraction: Ratio<u64>,
         last_emergency_restart: Option<EraId>,
         verifiable_chunked_hash_activation: EraId,
+        registry: &Registry,
```
Comment (Fraser999) [inline node/src/components/storage.rs]: I'd be inclined to make this optional (and hence also the `metrics`), so things like the tools don't need to import and construct a `Registry`.

File: node/src/components/storage.rs (hunk 1 above)
Comment (goral09) [reply node/src/components/storage.rs]: Yeah, we might for correctness. It shouldn't matter that much though b/c the error is fatal and we exit the process eventually.

File: node/src/components/storage.rs (hunk 3 above)
Comment (goral09) [reply node/src/components/storage.rs]: Sure, didn't think of that :( I'll make the change.

File: node/src/components/storage.rs (hunk 1 above)
Comment (marc-casperlabs) [reply node/src/components/storage.rs]: It'd be better, in my mind, to have two counters - one for started and one for finished requests. The in-flight number is then the difference between the two. This shows bugs in the counter earlier and gives us the actual rate for free!

File: node/src/components/storage/metrics.rs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+/// Bucket count, with the last bucket going to +Inf which will not be included in the results.
+/// - start = 1, factor = 3, count = 10
+/// - start * factor ^ count = 1 * 3.0 ^ 10 = 19683millis
+/// - Values above 19.683s will not fall in a bucket that is kept.
+const EXPONENTIAL_BUCKET_COUNT: usize = 10;
+
+#[derive(Debug)]
+pub(crate) struct Metrics {
+    pub(crate) sync_task_limiter_in_flight_counter: Counter,
```
Comment (marc-casperlabs) [inline node/src/components/storage/metrics.rs]: See comment - above. Let's create two metrics? :)

**Review by marc-casperlabs — COMMENTED** [review]
See comments about multiple metrics :).

File: node/src/components/storage.rs (hunk 1 above)
Comment (goral09) [reply node/src/components/storage.rs]: I'll agree to that but mainly b/c current approach doesn't work with debug releases :)

> This shows bugs in the counter earlier and gives us the actual rate for free!

Grafana is interpreting this data and calculates rates. I don't see a point in reinventing it here.

File: node/src/components/storage/metrics.rs (hunk 2 above)
Comment (goral09) [reply node/src/components/storage/metrics.rs]: Ah, forgot to update this one.

Comment (goral09) [reply node/src/components/storage/metrics.rs]: Addressed in [`bb58c84` (#3089)](https://github.com/casper-network/casper-node/pull/3089/commits/bb58c848473169841458f236d78596444da3817b)

File: node/src/components/storage.rs (hunk 3 above)
Comment (goral09) [reply node/src/components/storage.rs]: Addressed in [`51bf941` (#3089)](https://github.com/casper-network/casper-node/pull/3089/commits/51bf941ccac22112ce0a335fe798413f01ac15ce). Although it required a little bit of work so please review.

File: node/src/components/storage.rs (hunk 1 above)
Comment (goral09) [reply node/src/components/storage.rs]: @marc-casperlabs , addressed in [`7ab6c4e` (#3089)](https://github.com/casper-network/casper-node/pull/3089/commits/7ab6c4ef382e8051f56fb3548facb1917f16467d).

File: node/src/components/storage/metrics.rs [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        )?;
+
+        let sync_task_limiter_started_counter = Counter::new(
+            "storage_sync_task_limiter_started_count",
+            "number started tasks",
+        )?;
+        registry.register(Box::new(sync_task_limiter_started_counter.clone()))?;
+
+        let sync_task_limiter_completed_counter = Counter::new(
+            "storage_sync_task_limiter_completed_counter",
```
Comment (Fraser999) [inline node/src/components/storage/metrics.rs]:
```suggestion
            "storage_sync_task_limiter_completed_count",
```

**Review by Fraser999 — APPROVED** [review]
One nitpick to keep the metrics names' symmetric, but otherwise LGTM.

**Review by marc-casperlabs — APPROVED** [review]

**Pull request merged**